glk/osglk.c

```diff
@@ -517,13 +517,12 @@ void os_more_prompt()
 int os_askfile(const char *prompt, char *fname_buf, int fname_buf_len,
                int prompt_type, os_filetype_t file_type)
 {
+#ifndef GLKUNIX_FILEREF_GET_FILENAME
+    return OS_AFE_FAILURE;
+#else
```

Comment:
I slightly changed how the logic is done here too:

Originally this `#else` was an `#endif`, meaning the code following still was built (but never called). However, if `GLKUNIX_FILEREF_GET_FILENAME` isn't defined, then using `glkunix_fileref_get_filename()` is invalid. It's possible a compiler will optimize it away before the link phase, but it's not required, so it's best to just completely remove the code if it won't be called.

Also, I moved the variables into the `#else` block since they're unused otherwise, and this will prevent warnings with Glk implementations that don't provide `GLKUNIX_FILEREF_GET_FILENAME`.

Reply:
Yeah that's definitely the proper way to do it.